app/lib/feed_manager.rb

```diff
@@ -226,11 +226,11 @@ def phrase_filtered?(status, receiver_id, context)
   # added, and false if it was not added to the feed. Note that this is
   # an internal helper: callers must call trim or push updates if
   # either action is appropriate.
-  def add_to_feed(timeline_type, account_id, status)
+  def add_to_feed(timeline_type, account_id, status, aggregate_reblogs = true)
     timeline_key = key(timeline_type, account_id)
     reblog_key   = key(timeline_type, account_id, 'reblogs')
 
-    if status.reblog?
+    if status.reblog? && (aggregate_reblogs.nil? || aggregate_reblogs)
```

Comment:
This is a bit awkward, but basically none of the tests pass an actual local account (with an associated `User`) to those functions, so…